app/acapy.js

```diff
@@ -0,0 +1,86 @@
+const { ACAPY_API_KEY, ACAPY_ADMIN_SERVICE } = require('./env')
+
+const querySubWallets = async (subwalletName) => {
+  const fetch = (await import('node-fetch')).default
+
+  const url = new URL(`${ACAPY_ADMIN_SERVICE}/multitenancy/wallets`)
+  const search = new URLSearchParams({ wallet_name: subwalletName })
+  url.search = search.toString()
+
+  const queryResponse = await fetch(url, {
+    headers: {
+      'x-api-key': ACAPY_API_KEY,
+    },
+  })
+
+  const wallets = await queryResponse.json()
+
+  return wallets.results
+}
+
+const createSubWallet = async (body) => {
+  const fetch = (await import('node-fetch')).default
+
+  const url = new URL(`${ACAPY_ADMIN_SERVICE}/multitenancy/wallet`)
+  const createResponse = await fetch(url, {
+    method: 'POST',
+    headers: {
+      'x-api-key': ACAPY_API_KEY,
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify(body),
+  })
+
+  return createResponse
+}
+
+const getSubWalletToken = async (subWalletName) => {
+  const fetch = (await import('node-fetch')).default
+
+  const wallets = await querySubWallets(subWalletName)
+  if (wallets.length !== 1) {
+    return null
+  }
+
+  const wallet = wallets[0]
+  const url = new URL(`${ACAPY_ADMIN_SERVICE}/multitenancy/wallet/${wallet.wallet_id}/token`)
+  const subWalletToken = await fetch(url, {
+    method: 'POST',
+    headers: {
```

Comment:
maybe this could be converted into a class and headers included into a constructor? e.g.

```js
...
class AcapyWrapper {
  constructor(some_args) {
    this.headers = {
      'x-api-key': ACAPY_API_KEY,
      'content-type': 'application/json',
    };
    ...
  }
}
```

Reply:
There is some duplication here I could remove. The reason I haven't is that whilst these specific headers are duplicated twice there is a third instance in which an additional header is passed. I'm personally not a huge fan of `class` syntax (except in a few specfic cases, for example `Error` inheritance), but I could deduplicate this a bit.